src/uct/sm/mm/base/mm_iface.c

```diff
@@ -349,65 +343,67 @@ static uct_iface_ops_t uct_mm_iface_ops = {
     .iface_is_reachable       = uct_sm_iface_is_reachable
 };
 
-void uct_mm_iface_recv_desc_init(uct_iface_h tl_iface, void *obj, uct_mem_h memh)
+static void uct_mm_iface_recv_desc_init(uct_iface_h tl_iface, void *obj,
+                                        uct_mem_h memh)
 {
-    uct_mm_recv_desc_t *desc = obj;
-    uct_mm_seg_t *seg = memh;
-
-    /* every desc in the memory pool, holds the mm_id(key) and address of the
-     * mem pool it belongs to */
-    desc->key          = seg->mmid;
-    desc->base_address = seg->address;
-    desc->mpool_length = seg->length;
+    uct_mm_iface_t     *iface = ucs_derived_of(tl_iface, uct_mm_iface_t);
+    uct_mm_recv_desc_t *desc  = obj;
+    uct_mm_seg_t        *seg  = memh;
+    size_t offset;
+
+    if (seg->length > UINT_MAX) {
```

Comment:
ucs_unlikely()?

Reply:
it's slow path so not really needed here IMO